src/lib/FreeC/Backend/Agda/Converter/FuncDecl.hs

```diff
@@ -31,22 +31,17 @@ convertFuncDecl decl = localEnv $ sequence [convertSignature decl]
 -- | Converts the type signature of the given function to an Agda type
 --   declaration.
 convertSignature :: IR.FuncDecl -> Converter Agda.Declaration
-convertSignature (IR.FuncDecl _ ident tVars args retType _) =
-  Agda.funcSig <$> lookupValueIdent ident <*> convertFunc_ tVars types retType
-  where types = map IR.varPatType args
-
--- | Looks up the name of a Haskell function in the environment and converts it
---   to an Agda name.
-lookupValueIdent :: IR.DeclIdent -> Converter Agda.Name
-lookupValueIdent (IR.DeclIdent srcSpan name) =
-  Agda.unqualify <$> lookupAgdaIdentOrFail srcSpan IR.ValueScope name
+convertSignature (IR.FuncDecl _ (IR.DeclIdent srcSpan name) tVars args retType _)
+  = Agda.funcSig <$> ident <*> convertFunc_ tVars types retType
+ where
+  types = map IR.varPatType args
+  ident = lookupUnQualAgdaIdentOrFail srcSpan IR.ValueScope name
 
 -- | Converts a fully applied function.
 convertFunc_
   :: [IR.TypeVarDecl] -> [Maybe IR.Type] -> Maybe IR.Type -> Converter Agda.Expr
```

Comment:
Maybe that's just me not knowing the code well enough, but was such a naming scheme used before? It's first time I stumble upon a suffix `_`.

Reply:
The Agda code base uses this naming structure sometimes for  specialized versions of the same function (e.g. `mkBinder_` and `mkBinder`, `mkBoundName_` and `mkBoundName`). I use this notation for some smart constructors in `FreeC.Backend.Agda.Syntax`.
In this case the name is probably just a result of renaming and moving functions arround. I will change it.